sawmill-core/src/main/java/io/logz/sawmill/processors/DeDotProcessor.java

```diff
@@ -0,0 +1,82 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@ProcessorProvider(type = "deDot", factory = DeDotProcessor.Factory.class)
+public class DeDotProcessor implements Processor {
+    private String separator;
+
+    public DeDotProcessor(String separator) {
+        this.separator = separator;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) throws InterruptedException {
+        Map<String,Object> docAsMap = doc.getSource();
+        doc.replace(deDotMap(docAsMap));
+        return ProcessResult.success();
+    }
+
+
+    private Map<String,Object> deDotMap(Map<String, Object> docAsMap) throws InterruptedException {
+
+        Map<String,Object> mapClone = new LinkedHashMap<>();
+        for (Map.Entry<String, Object> entry : docAsMap.entrySet()) {
+            if (Thread.currentThread().isInterrupted()) throw new InterruptedException();
+            String dedotedKey = deDotKey(entry.getKey());
+            if (entry.getValue() instanceof Map) {
+                Map<String, Object> deDotedMap = deDotMap((Map) entry.getValue());
+                mapClone.put(dedotedKey, deDotedMap);
```

Comment:
in all the cases you need to put the new key and value,
just do it after all conditions